pkg/api/v1alpha1/vspheremachineconfig.go

```diff
@@ -76,3 +78,23 @@ func GetVSphereMachineConfigs(fileName string) (map[string]*VSphereMachineConfig
 	}
 	return configs, nil
 }
+
+func validateVSphereMachineConfig(config *VSphereMachineConfig) error {
+	if len(config.Spec.Datastore) <= 0 {
+		return fmt.Errorf("VSphereMachineConfig %s datastore is not set or is empty", config.Name)
+	}
+	if len(config.Spec.Folder) <= 0 {
+		logger.Info("VSphereMachineConfig folder is not set or is empty. Will default to root vSphere folder.")
```

Comment:
I'm not sure this log belongs here, could we find a better place? For starters this is talking about some defaulting logic, while this method runs validations

Reply:
I'll move it to the method where we set defaults. That will be in a separate PR coming soon